### PR TITLE
Added examples to Terra-Text for Date and Time format

### PR DESCRIPTION
**Pull request by swetasingh9714**
### Summary
Added examples to Terra-Text for Date and Time format.

Screenshot 2023-04-13 at 4 03 55 PM

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-core-docs/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Fixed
+  * UI text for date and hour format to be read correctly by the screen readers.
+
```
Comment (sdadn) [inline packages/terra-core-docs/CHANGELOG.md:7]:
```suggestion

* Added
  * Additional examples for `terra-text` to demonstrate valid date and time formats.
  
```

File: packages/terra-text/src/Text.jsx [hunk 2]
```diff
@@ -96,9 +97,49 @@ const Text = ({
...
+        ? (<>
+          <span className={descriptiveClasses}>Hour</span>
+          <React.Fragment>{hourVal}:</React.Fragment>
+
+          <span className={descriptiveClasses}>Minute</span>
+          <React.Fragment>{minuteVal}:</React.Fragment>
+
+          <span className={descriptiveClasses}>Second</span>
+          <React.Fragment>{secondVal}</React.Fragment>
+        </>) : children}
```
Comment (sdadn) [inline packages/terra-text/src/Text.jsx:142]: There needs to be unit & WDIO tests for the updated logic.

Comment (swetasingh9714) [reply packages/terra-text/src/Text.jsx]: @sdadn 
There are some changes in the code. All the logic is shifted to examples and no API changes are done for this.

File: package.json [hunk 3]
```diff
@@ -121,5 +121,8 @@
     "webpack-cli": "^4.6.0",
     "webpack-dev-server": "^4.11.1",
     "webpack-merge": "^5.1.4"
+  },
+  "dependencies": {
+    "moment": "^2.29.4"
```
Comment (supreethmr) [inline package.json:126]: let's move this to `devDependencies`

File: packages/terra-core-docs/src/terra-dev-site/test/text/DateTest.test.jsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+const descriptiveClasses = classNames(
+  cx([
+    'text',
+    { 'visually-hidden': true },
+  ]),
+);
+
+const dateVar = moment('01-10-2022').format('MM/DD/YYYY');
+
+const isValidDate = moment(dateVar, 'MM/DD/YYYY', true).isValid();
```
Comment (supreethmr) [inline packages/terra-core-docs/src/terra-dev-site/test/text/DateTest.test.jsx:18]: Let's remove this as dateVar will always have a valid date.

File: packages/terra-core-docs/src/terra-dev-site/test/text/TimeTest.test.jsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+const descriptiveClasses = classNames(
+  cx([
+    'text',
+    { 'visually-hidden': true },
+  ]),
+);
+
+const timeVar = moment('10:22:30', 'hh:mm:ss').format('hh:mm:ss');
+
+const isValidTime = moment(timeVar, 'hh:mm:ss', true).isValid();
```
Comment (supreethmr) [inline packages/terra-core-docs/src/terra-dev-site/test/text/TimeTest.test.jsx:18]: same here I think we don't need validation as time value will always be valid

**Comment by supreethmr** [comment]
[packages/terra-core-docs/src/terra-dev-site/test/text/DateTest.test.jsx](https://github.com/cerner/terra-core/pull/3777/files/30f54551c748ff894fa7e6da2c4f2032da2a980f#diff-8ca3c4271cd49dba8e41490942de1ef426a1855408c15309347311c9619508d7) this file should be added as doc examples not as test files. because test files are not visible for consumers on terra-ui site. 

so I would suggest you to move both Date and Time examples to https://github.com/cerner/terra-core/tree/30f54551c748ff894fa7e6da2c4f2032da2a980f/packages/terra-core-docs/src/terra-dev-site/doc/text/example

File: packages/terra-core-docs/src/terra-dev-site/test/text/DateTest.test.jsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
```
Comment (ShettyAkarsh) [inline packages/terra-core-docs/src/terra-dev-site/test/text/DateTest.test.jsx:1]: shouldn't this be added as part of examples. instead of tests ?

**Comment by ShettyAkarsh** [comment]
> ### Summary
> Changes made to Text.jsx to fix the issue : UI text for date and hour format is not read correctly by the screen readers
> 
> Thank you for contributing to Terra. @cerner/terra

did we fixed the terra-text source code here ? can we update the pr description in relation to what has been changed. even the title ?

**Comment by ShettyAkarsh** [comment]
@swetasingh9714 , can you add the respective screenshot here on how the new examples looks like after this change ? and make sure we have tested the same with default and fusion theme.

File: packages/terra-core-docs/src/terra-dev-site/test/text/TimeTest.test.jsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import Text from 'terra-text';
+import moment from 'moment';
+import classNames from 'classnames';
+import classNamesBind from 'classnames/bind';
+import styles from '../../../../../terra-text/src/Text.module.scss';
+
+const cx = classNamesBind.bind(styles);
```
Comment (supreethmr) [inline packages/terra-core-docs/src/terra-dev-site/test/text/TimeTest.test.jsx:8]: same here !! we can remove the test file and keep only examples

File: packages/terra-core-docs/src/terra-dev-site/doc/text/example/TextTime.jsx [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+const timeVar = moment('10:22:30', 'hh:mm:ss').format('hh:mm:ss');
+
+const timeParts = timeVar.split(':');
+const hourVal = timeParts[0];
+const minuteVal = timeParts[1];
+const secondVal = timeParts[2];
+
+const valueToDisplay = (
+  <>
+    <span className={descriptiveClasses}>Hour</span>
```
Comment (supreethmr) [inline packages/terra-core-docs/src/terra-dev-site/doc/text/example/TextTime.jsx:25]: is there any specific reason for having this classname on labels ??

Comment (swetasingh9714) [reply packages/terra-core-docs/src/terra-dev-site/doc/text/example/TextTime.jsx:25]: This adds the necessary styling to the label/span so as to not visually show it on the screen, and instead only be perceptible to the screen readers.

File: packages/terra-core-docs/src/terra-dev-site/doc/text/example/TextDate.jsx [hunk 9]
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import Text from 'terra-text';
+import moment from 'moment';
+import classNames from 'classnames';
+import classNamesBind from 'classnames/bind';
+import styles from '../../../../../../terra-text/src/Text.module.scss';
```
Comment (supreethmr) [inline packages/terra-core-docs/src/terra-dev-site/doc/text/example/TextDate.jsx:6]:
```suggestion
import VisuallyHiddenText from 'terra-visually-hidden-text';
```

File: packages/terra-core-docs/src/terra-dev-site/doc/text/example/TextTime.jsx [hunk 10]
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import Text from 'terra-text';
+import moment from 'moment';
+import classNames from 'classnames';
+import classNamesBind from 'classnames/bind';
+import styles from '../../../../../../terra-text/src/Text.module.scss';
```
Comment (supreethmr) [inline packages/terra-core-docs/src/terra-dev-site/doc/text/example/TextTime.jsx:6]:
```suggestion
import VisuallyHiddenText from 'terra-visually-hidden-text';
```

File: packages/terra-core-docs/CHANGELOG.md [hunk 11]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Added
+  * Additional examples for `terra-text` to demonstrate valid date and time formats.
```
Comment (supreethmr) [inline packages/terra-core-docs/CHANGELOG.md:6]:
```suggestion
  * Added examples for `terra-text` to demonstrate valid date and time formats.
```

**Review by sdadn — APPROVED** [review]

**Review by supreethmr — APPROVED** [review]

**Comment by supreethmr** [comment]
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time
https://developer.mozilla.org/en-US/docs/Web/API/HTMLTimeElement/dateTime

Can we update the examples to html semantic elements <time> <dateTime> as shown on above links instead of using labels to identify them !!

**Review by swetasingh9714 — COMMENTED** [review]
@neilpfeiffer I have updated the code to use semantic element. Please take a look when you find time. We are hoping to close this by EOD.

Thank you!

**Review by supreethmr — APPROVED** [review]

**Review by saket2403 — APPROVED** [review]

**Pull request merged**